packages/babel-helper-transform-fixture-test-runner/src/index.js

```diff
@@ -386,11 +397,28 @@ function run(task) {
       fs.statSync(path.dirname(expected.loc)).isDirectory() &&
       !process.env.CI
     ) {
-      console.log(`New test file created: ${expected.loc}`);
-      fs.writeFileSync(expected.loc, `${result.code}\n`);
+      const expectedFile = expected.loc.replace(
+        /\.m?js$/,
+        result.sourceType === "module" ? ".mjs" : ".js",
+      );
+
+      console.log(`New test file created: ${expectedFile}`);
+      fs.writeFileSync(expectedFile, `${result.code}\n`);
+
+      if (expected.loc !== expectedFile) {
+        try {
+          fs.unlinkSync(expected.loc);
```

Comment:
I don't understend when this code is executed. If `!expected.code` (line 394), shouldn't `output.m?js` not exist?

Reply:
It's not super likely, so that's a fair question. Technically the file could have been emptied instead of deleted, so I figured it wouldn't hurt to add some logic to try to avoid creating sibling files.